In [21]:
from scipy import stats
from pandas import read_excel
import pandas as pd

In [22]:
df = read_excel("https://data.hossam.kr/E02/material.xlsx")
df

상표1   상표2
0   80.4  80.1
1   78.2  81.2
2   80.1  79.5
3   77.1  78.0
4   79.6  76.1
5   80.4  77.0
6   81.6  80.1
7   79.9  79.9
8   84.4  78.8
9   80.9  80.8
10  83.1   NaN

In [23]:
df.describe()

상표1        상표2
count  11.000000  10.000000
mean   80.518182  79.150000
std     2.038047   1.659485
min    77.100000  76.100000
25%    79.750000  78.200000
50%    80.400000  79.700000
75%    81.250000  80.100000
max    84.400000  81.200000

In [24]:
df1 = df.fillna(79.1)
df1

상표1   상표2
0   80.4  80.1
1   78.2  81.2
2   80.1  79.5
3   77.1  78.0
4   79.6  76.1
5   80.4  77.0
6   81.6  80.1
7   79.9  79.9
8   84.4  78.8
9   80.9  80.8
10  83.1  79.1

귀무가설 - 상표 1과 상표 2의 각 원료의 함량은 다르지 않다.

대립가설 - 상표 1과 상표 2의 각 원료의 함량은 다르다.

In [25]:
stats.ttest_rel(df1['상표1'], df1['상표2'], alternative='less')

TtestResult(statistic=1.81443733772276, pvalue=0.9501609790909215, df=10)

결과 해석
###### 대응표본 검정결과에서 검정통계랑 값이 1.814이고 유의확률 P값이 0.95로 유의 수준 0.05보다 크므로 상표 1과 상표 2의 각 원료의 함량이 다르다는 귀무가설을 기각할 수 없다. 즉, 주성분 A의 함량이 다르지 않다.

문제 2

In [26]:
df = read_excel("https://data.hossam.kr/E02/blood_pressure.xlsx")
df

복용전  복용후
0    70   68
1    80   72
2    72   62
3    76   70
4    76   58
5    76   66
6    72   68
7    78   52
8    82   64
9    64   72
10   74   74
11   92   60
12   74   74
13   68   72
14   84   74

귀무가설 - 복용전보다 복용 후에 혈압이 차이가 없다

대립가설 - 복용전보다 복용 후에 혈압이 차이가 있다.

In [27]:
stats.ttest_rel(df['복용전'], df['복용후'], alternative='two-sided')

TtestResult(statistic=3.105360487466109, pvalue=0.007749436106654119, df=14)

결과 해석
###### 대응표본 검정결과에서 검정통계랑 값이 3.10이고 유의확률 P값이 0.0077로 유의 수준 0.05보다 작기 때문에 복용전보다 복용 후에 혈압 차이가 없다는 귀무가설을 기각할 수 없다.. 즉, 복용전과 후에 혈압에는 차이가 없다. 

문제3

In [28]:
df = read_excel("https://data.hossam.kr/E02/runner_diet.xlsx")
df

감량전   감량후    
0  14.5  14.3 NaN
1  13.4  13.4 NaN
2  13.0  13.1 NaN
3  14.5  13.9 NaN
4  13.3  13.1 NaN
5  14.2  13.9 NaN
6  13.6  13.6 NaN
7  13.9  14.0 NaN
8  13.2  12.8 NaN
9  13.0  12.8 NaN

In [29]:
df.isnull().sum()

감량전     0
감량후     0
       10
dtype: int64

In [30]:
df1 = df.dropna(axis=1)
df1

감량전   감량후
0  14.5  14.3
1  13.4  13.4
2  13.0  13.1
3  14.5  13.9
4  13.3  13.1
5  14.2  13.9
6  13.6  13.6
7  13.9  14.0
8  13.2  12.8
9  13.0  12.8

In [31]:
A = [1,2,3,4,5,6,7,8,9,10]
df3 = pd.DataFrame(A)
df2 = pd.concat([df1, df3], axis=1)
print(df2)

    감량전   감량후   0
0  14.5  14.3   1
1  13.4  13.4   2
2  13.0  13.1   3
3  14.5  13.9   4
4  13.3  13.1   5
5  14.2  13.9   6
6  13.6  13.6   7
7  13.9  14.0   8
8  13.2  12.8   9
9  13.0  12.8  10


In [32]:
df2.columns

Index(['감량전', '감량후', 0], dtype='object')

In [33]:
df2.columns = ['감량전', '감량후', '선수번호']
df4 = df2.set_index('선수번호')
df4

감량전   감량후
선수번호            
1     14.5  14.3
2     13.4  13.4
3     13.0  13.1
4     14.5  13.9
5     13.3  13.1
6     14.2  13.9
7     13.6  13.6
8     13.9  14.0
9     13.2  12.8
10    13.0  12.8

귀무가설 - 감량전보다 감량후가 달리기의 속도가 차이가없다.  

대립가설 - 감량전보다 감량후가 달리기의 속도가 차이가있다..

In [40]:
stats.ttest_rel(df4['감량전'], df4['감량후'], alternative='two-sided')

TtestResult(statistic=2.375306807446375, pvalue=0.04154604923393679, df=9)

###### 대응표본 검정결과에서 검정통계랑 값이 2.37이고 유의확률 P값이 0.041로 유의 수준 0.05보다 작기 때문에 감량전보다 감량후의 달리기 속도가 차이가없다는 귀무가설을 기각할 수없다. 즉, 체중을 감량하는 것은 달리기 속도에 영향을 미친다.

문제 4

In [35]:
df = read_excel("https://data.hossam.kr/E02/wrestler_diet.xlsx")
df

이름  감량전 악력  감량후 악력  감량전 윗몸일으키기  감량후 윗몸일으키기  감량전 턱걸이  감량후 턱걸이
0   B.S      43      41          35          41       25       29
1   P.J      42      41          40          44       25       29
2   G.G      52      50          36          41       22       23
3   A.L      53      52          38          42       10       15
4   I.H      44      40          36          42       15       19
5   B.Y      46      47          39          44       17       19
6   I.K      48      47          35          42       24       26
7   K.B      58      55          34          36       17       18
8   K.J      50      50          33          35       17       20
9   S.S      48      50          39          45       21       25
10  S.P      56      48          37          37       31       35
11  S.J      43      41          33          37       15       22

In [36]:
stats.ttest_rel(df['감량전 악력'], df['감량후 악력'], alternative='less')

TtestResult(statistic=2.3654081485744824, pvalue=0.9812727768474901, df=11)

비교적 감량 후 떨어지는 것을 확인할 수 있다.

In [37]:
stats.ttest_rel(df['감량전 윗몸일으키기'], df['감량후 윗몸일으키기'], alternative='greater')

TtestResult(statistic=-7.179926973956724, pvalue=0.9999910088194148, df=11)

감량 후 비교적 윗몸일으키기 숫자가 늘어남을 확인할 수 있다.

In [38]:
stats.ttest_rel(df['감량전 턱걸이'], df['감량후 턱걸이'], alternative='greater')

TtestResult(statistic=-6.841977654731714, pvalue=0.9999860314894063, df=11)

감량 후 비교적 턱걸이 숫자가 늘어남을 확인할 수 있다.

##### 반복문을 적용하여 풀어보기

In [41]:
df2 = df.set_index('이름')
cols = df2.columns
cols

Index(['감량전 악력', '감량후 악력', '감량전 윗몸일으키기', '감량후 윗몸일으키기', '감량전 턱걸이', '감량후 턱걸이'], dtype='object')

In [42]:
n = len(cols)
n

6

In [43]:
for i in range(0, n, 2):
    print(i, i+1)
    result = stats.ttest_rel(df2[df2.columns[i]], df2[df2.columns[i+1]], alternative='two-sided')
    print(result)

0 1
TtestResult(statistic=2.3654081485744824, pvalue=0.037454446305019644, df=11)
2 3
TtestResult(statistic=-7.179926973956724, pvalue=1.798236117030195e-05, df=11)
4 5
TtestResult(statistic=-6.841977654731714, pvalue=2.7937021187235204e-05, df=11)
